tests/meta/archive/test_archiving.py

```diff
@@ -0,0 +1,27 @@
+from pathlib import Path
+from typing import Dict, List
+
+from cg.meta.archive.ddn_dataflow import DDNDataFlowApi, TransferData
+
+
+def test_correct_destination_root(
+    ddn_api: DDNDataFlowApi, local_directory: str, transfer_data: TransferData
+):
+    """Tests the formatting function of the path dict for retrieving."""
+    # GIVEN a source path and a destination path
+
+    # WHEN creating the correctly formatted dictionary
+    transfer_data.correct_destination_root()
+
+    # THEN the destination path should be the local directory minus the /home part
+    assert transfer_data.destination == (Path(*local_directory.parts[2:]))
+
+
+def test_add_repositories(ddn_config, local_directory, remote_directory, transfer_data):
+    transfer_data.add_repositories(
+        source_prefix=ddn_config.local_storage, destination_prefix=ddn_config.archive_repository
+    )
+    assert transfer_data.source == ddn_config.local_storage + str(local_directory)
+
+
+# TODO add more tests and clean them up
```

Comment:
Here are some test cases that we could consider:

### Initialization
- Test if the DDNDataFlowApi class initializes correctly with valid DDNConfig inputs.
- Test if the DDNDataFlowApi class raises appropriate errors with invalid DDNConfig inputs.

### Authentication
- Test if the _set_auth_tokens method successfully retrieves and sets authentication and refresh tokens.
- Test if the _refresh_auth_token method successfully updates the authentication token when it expires.
- Test how the class handles invalid credentials.

### TransferData class
- Test if the correct_source_root method correctly modifies the source root path.
- Test if the correct_destination_root method correctly modifies the destination root path.
- Test if the add_repositories method correctly adds source and destination prefixes.

### TransferPayload class
- Test if the correct_source_root method correctly modifies the source root path for all TransferData objects.
- Test if the correct_destination_root method correctly modifies the destination root path for all TransferData objects.
- Test if the add_repositories method correctly adds source and destination prefixes for all TransferData objects.
- Test if the dict method returns a dictionary with the expected structure and content.

### Archiving folders

- Test if the archive_folders method successfully archives folders for valid input (sources_and_destinations dictionary).
- Test if the archive_folders method returns the expected response for invalid input.
- Test if the archive_folders method handles exceptions, such as when a source folder does not exist, or if the destination path is invalid.


### Retrieving folders

- Test if the retrieve_folders method successfully retrieves folders for valid input (sources_and_destinations dictionary).
- Test if the retrieve_folders method returns the expected response for invalid input.
 Test if the retrieve_folders method handles exceptions, such as when a source folder does not exist in the archive, or if the destination path is invalid.


### Edge cases:

- Test archiving and retrieving folders with special characters in their names.
- Test archiving and retrieving folders with a large number of files or a large total size.
- Test archiving and retrieving folders with various file types and permissions.
- Test archiving and retrieving folders with deeply nested structures.